### PR TITLE
Updating documentation to reflect the latest status of extension resources

### PR DESCRIPTION
**Pull request by nikhiljindal**
Follow up of https://github.com/kubernetes/kubernetes/pull/15516

cc @bgrant0607 @erictune @lavalamp

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -107,11 +107,12 @@ In the future we expect that there will be more API groups, all at REST path `/a
...
-## Enabling the extensions group
+## Enabling resources in the extensions group
 
-Enable `extensions/v1beta1` objects by adding the following flags to your API server:
-
-  - `--runtime-config=extensions/v1beta1=true`
+Jobs, Ingress and HorizontalPodAutoscalers are enabled by default.
+Other extensions resources can be enabled by setting runtime-config on
+apiserver. For ex: to enable deployments, set
+`--runtime-config=extensions/v1beta1/deployments=true`
```
Comment (lavalamp) [inline docs/api.md]: Is there a way to see what the default is?

Comment (nikhiljindal) [inline docs/api.md]: Didnt get the question.
Default for runtime-config is empty.
Resources enabled by default are jobs, ingress and hpa.

Comment (lavalamp) [inline docs/api.md]: Right, I understand. But suppose I have an arbitrary apiserver binary, how do I learn that? Is the only way to start it and see what it serves?

Comment (erictune) [inline docs/api.md]: I think we are moving away from flag-based defaults, as we move toward #13146 
Maybe @mikedanese has thoughts about how users can learn defaults when using versioned config.  At any rate, seems like this is a useful docs improvement regardless of the answer.

Comment (nikhiljindal) [inline docs/api.md]: Yes, starting the server and seeing what is served is the only way I can think of.

Comment (paralin) [inline docs/api.md]: How can you set multiple?

Comment (nikhiljindal) [inline docs/api.md]: Its csv

Comment (bgrant0607) [inline docs/api.md]: The API discovery mechanism is the right way to determine which APIs are exposed.

Comment (paralin) [inline docs/api.md]: Works, thanks @nikhiljindal

Comment (lavalamp) [inline docs/api.md]: We should state here that it's comma-separated, or link to the documentation.

**Comment by mikedanese** [comment]
Is there a way to enable the entire API Group?

File: docs/api.md (hunk 1 above)
Comment (nikhiljindal) [inline docs/api.md]: Done

**Comment by nikhiljindal** [comment]
We do not have ```extensions/v1beta1/all=true```, but we have talked about adding it.
We are anyway, going to enable all extensions resources by default on HEAD.

**Comment by nikhiljindal** [comment]
Pushed a new version as per comments. PTAL

**Comment by paralin** [comment]
If you're enabling all extensions on default then my PR removing the env
vars to enable experimental can be safely pulled as ENABLE_DEPLOYMENTS
becomes irrelevant

**Comment by lavalamp** [comment]
We should make `extensions/v1beta1=true` turn on everything,
`extensions/v1beta1=false` turn off everything, and have the absence of
mention leave all the defaults.

Also we should make the default print when you do `kube-apiserver --help`
but that's an issue for another day.

**Comment by nikhiljindal** [comment]
Actually, I have asked for a clarification on when can we enable deployments by default (there seem to be some confusion): https://github.com/kubernetes/kubernetes/issues/15313#issuecomment-150327432

Have also filed an issue for better support for enabling all resources: https://github.com/kubernetes/kubernetes/issues/16117.

@erictune any comments on this PR or can I get an LGTM? I need to cherrypick this into 1.1 as well.

**Comment by erictune** [comment]
lgtm

**Comment by saad-ali** [comment]
All tests passed so manually merging to drain merge queue.

**Pull request merged**